packages/api-server/src/requestHandlers/awsLambdaFastify.ts

```diff
@@ -39,6 +48,57 @@ const lambdaEventForFastifyRequest = (
   } as APIGatewayProxyEvent
 }
 
+/**
+ * The header keys are case insensitive, but Fastify prefers these to be lowercase.
+ * Therefore, we want to ensure that the headers are always lowercase and unique
+ * for compliance with HTTP/2.
+ *
+ * @see: https://www.rfc-editor.org/rfc/rfc7540#section-8.1.2
+ *
+ * Just as in HTTP/1.x, header field names are strings of ASCII
+ * characters that are compared in a case-insensitive fashion.  However,
+ * header field names MUST be converted to lowercase prior to their
+ * encoding in HTTP/2.  A request or response containing uppercase
+ * header field names MUST be treated as malformed (Section 8.1.2.6).
+ *
+ * @see: https://tools.ietf.org/html/rfc2616#section-4.2
+ */
+const sanitizeHeaders = (
+  headers: FastifyLambdaHeaders,
+  reply: FastifyReply
+) => {
+  const headersLowercase = {} as FastifyLambdaHeaders
+
+  if (headers && headersLowercase) {
+    reply.headers({})
+
+    Object.keys(headers).forEach((headerName) => {
+      headersLowercase[headerName.toLowerCase()] = headers[headerName]
+    })
+
+    reply.headers(headersLowercase)
+  }
+}
+
+/**
+ * In case there are multi-value headers that are not in the headers object,
+ * we need to add them to the headers object and ensure the header names are lowercase
+ * and that if there is an array fo values, they are separated by a semi-colon.
+ */
+const mergeMultiValueHeaders = (
```

Comment:
Are you sure there are cases where multivalue headers are present and headers aren't? 

My understanding is that MVH are duplicated - its just a proxy setting.

Reply:
Resolving as this is definitely required!